yt/frontends/chimera/data_structures.py

```diff
@@ -0,0 +1,295 @@
+"""
+Chimera data structures
+
+
+
+"""
+
+import os
+
+import numpy as np
+
+from yt.data_objects.index_subobjects.unstructured_mesh import SemiStructuredMesh
+from yt.data_objects.static_output import Dataset
+from yt.geometry.geometry_handler import YTDataChunk
+from yt.geometry.unstructured_mesh_handler import UnstructuredIndex
+from yt.utilities.file_handler import HDF5FileHandler
+from yt.utilities.io_handler import io_registry
+from yt.utilities.on_demand_imports import _h5py as h5py
+
+from .fields import ChimeraFieldInfo
+
+
+class ChimeraMesh(SemiStructuredMesh):
+
+    _index_offset = 0
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+
+def _find_files(
+    filename_c,
+):  # Returns a list of all files that share a frame number with the input
+    dirname, file = os.path.split(filename_c)
+    dir_contents = os.listdir(dirname)
+    frames = [f for f in dir_contents if file[:-7] in f]
+    directory = os.path.dirname(filename_c)
+    index_filenames = [directory + os.path.sep + f for f in frames]
+    index_filenames.sort()
+    return index_filenames
```

Comment:
Here's a direct suggestion to make this function more compact (elegant ?). It doesn't resolve the issue with `file[:-7]` yet. Since you clarified in a previous comment that the intention is *not* to split the extension, can you show what the last 7 chars are supposed to look like so I can suggest how to express it more explicitly in the code ?
```suggestion
def _find_files(filename_c):  
    # Returns a list of all files that share a frame number with the input
    dirname, file = os.path.split(filename_c)
    frames = [f for f in os.listdir(dirname) if file[:-7] in f]
    index_filenames = [os.path.join(dirname, f) for f in sorted(frames)]
    return index_filenames
```

Reply:
Sure, the standard form of the file names is along the lines of `chimera_02050_grid_1_01.h5` the `grid_1` exists for that Yin-Yang data (with 2 grids). Each 3D grid is stored as several files, with the `01` indexing these portions of the grid. files[:-7] matches all files with the same common first section but for all grid numbers and file index numbers.

Reply:
Cool, I'll edit my suggestion above then

Reply:
done. Note that `import re` should be also added at the top level of the module

Reply:
I incorporated this method, with the slight alteration that the frame number are not of uniform length, so `\d+` is more appropriate than `\d{5}`

Reply:
makes sense, thank you!